### ISSUE TITLE
How to set the parameter of variance_fraction in the iucnn_prepare_phy?

### ISSUE THREAD

**Issue by Orange-chen-PEAR** [IUCNN/IUCNN#47]
Hi @all,

I had some trouble in selecting number of Eigenvalues that cumulative explain a specified fraction of variance. When I use different values of variance_fraction, I'll get different columns (extracting phylogenetic eigenvectors as NN features) of matrix, which may affect the results of training a IIUCNN model.

I want to know the criterions or references for determining the variance_fraction. 
Can you help me with this?
Thank you for your kind consideration of this request.

Jinnan 

For example:
`library(IUCNN)`
`library(ape)`
`tree <- rphylo(n = 10, birth=0.1, death=0)`
`iucnn_prepare_phy(phy = tree, variance_fraction = 0.6)`

```
species   phylo c1    phylo c2
1       t1 -0.1380841  0.01297110
2       t2 -0.2435313  0.04797561
3       t3 -0.2479926  0.04935292
4       t4 -0.1504452 -0.00658140
5       t5  0.3585364 -0.87214856
6       t6  0.5317430  0.33884160
7       t7  0.5317430  0.33884160
8       t8 -0.1504452 -0.00658140
9       t9 -0.2479926  0.04935292
10     t10 -0.2435313  0.04797561
```

`iucnn_prepare_phy(phy = tree, variance_fraction = 1)`
```
species   phylo c1    phylo c2    phylo c3    phylo c4     phylo c5      phylo c6      phylo c7      phylo c8
1       t1 -0.1380841  0.01297110 -0.01619169  0.93823535  0.014723485  2.202402e-17 -1.433708e-16  0.000000e+00
2       t2 -0.2435313  0.04797561 -0.25126267 -0.13948718 -0.505767987 -7.071068e-01  7.461978e-16  1.395698e-16
3       t3 -0.2479926  0.04935292 -0.26008693 -0.15600492  0.494045743  4.475587e-16  7.071068e-01 -8.717601e-17
4       t4 -0.1504452 -0.00658140  0.60276604 -0.11827388  0.002703725 -2.498002e-16 -2.292108e-16  6.347164e-16
5       t5  0.3585364 -0.87214856 -0.09446000 -0.04340804  0.001250476  1.110223e-16 -1.577517e-17 -2.583407e-18
6       t6  0.5317430  0.33884160 -0.03609058 -0.03364768  0.001031540 -5.551115e-17 -4.420079e-16  7.071068e-01
7       t7  0.5317430  0.33884160 -0.03609058 -0.03364768  0.001031540 -5.551115e-17  3.351482e-16 -7.071068e-01
8       t8 -0.1504452 -0.00658140  0.60276604 -0.11827388  0.002703725  3.191891e-16  2.426340e-16 -5.470987e-16
9       t9 -0.2479926  0.04935292 -0.26008693 -0.15600492  0.494045743 -1.547373e-15 -7.071068e-01 -3.751356e-16
10     t10 -0.2435313  0.04797561 -0.25126267 -0.13948718 -0.505767987  7.071068e-01 -1.151208e-15  2.892323e-16

**Comment by dsilvestro** [comment]
Hi thank you for bringing this up. The amount of variance explained by the eigenvectors will increase the more values (phylogenetic features in the NN) you include. If you want to include the entire phylogenetic signal (i.e. `variance_fraction = 1`) then this will result potentially in a lot of additional features. However, usually there first few axes (columns) contain most of the phylogenetic information and selecting a `variance_fraction` < 1 can be a way to avoid giving too much weight to phylogenetic features with respect to other features. 

The value is however dataset specific and the best way to deal with that is to tune the variance_fraction as a hyper-parameter in the model, just like the number of layers and nodes in the NN. That means, try a couple of different values and select the model returning the highest cross-validation accuracy (or lowest CV loss).  

I hope this helps!
Daniele
